prefixing.graphql

```diff
@@ -0,0 +1,54 @@
+schema
+  @using(spec: "https://spec.example.com/specA/v1.0")
+  @using(spec: "https://spec.example.com/specB/v2.0", prefix: "B") {
+  query: Query
+}
+
+"""
+specA_SomeType is a type defined by spec A.
+"""
+type specA_SomeType {
+  """
+  nativeField is a field defined by specA on a type also defined
+  by specA (namely specA_SomeType)
+  """
+  nativeField: Int @specA_fieldDirective
+
+  """
+  B_foreignField is a field defined by specB on a type defined
+  by specA (namely specA_SomeType)
+  """
+  B_foreignField: Int
```

Comment:
It's a bit surprising to see `B` fields on a `specA` type without a previous mention of the possible interdependence of specs. Should there be a section that clarifies this?

Am I understanding correctly that in order to use `specB`, you must also use `specA` in this example? Is there a way we can make that clear to processors so they can make smart decisions around validating a document with spec interdependence like this?

Reply:
There is text in the prefixing section re: prefixing fields on foreign types / arguments to foreign directives—

> Specs **should** attach their prefix to:
> ...
> - the names of any fields the spec introduces on *foreign types* defined in a different spec
> - the names of any arguments the spec introduces on *foreign directives and fields* defined in a different spec

It may be the case that to use `specB` you must also use `specA`. It may also be the case that `specB` does not require `specA`, but *if* you are using `specA`, `specB` provides some additional fields on `specA`'s types (kindof a soft peer dependency). You can't know the situation without reading the specs.

Expressing spec dependencies in the `@using` directive could be cumbersome (imagine a whole transitive tree), and there would be no way for us to validate whether the dependencies in the document are correct, so it seems a bit like kicking the can. Generally, I think it's better to leave such validations to the individual specs (i.e. the composed schema spec can have a validation that the document is `@using` an appropriate version of it; some hypothetical `gslb` spec could validate that there's a reference to an appropriate version of composed schema if it requires that to function, etc.) This means we cannot validate that all the necessary specs are around if we don't know all the specs in the document, but that seems fine?